src-tauri/src/bin/ts-rs-export.rs

```diff
@@ -0,0 +1,67 @@
+use std::{
+    fs,
+    path::Path,
+    process::{exit, Command},
+};
+
+use jute::backend::notebook::Notebook;
+use ts_rs::TS;
+
+fn main() {
+    let export_path = Path::new("../src/bindings");
+
+    // print the full path of the export directory
+    println!(
+        "Exporting TypeScript bindings to `{:?}`",
+        fs::canonicalize(export_path).expect("Failed to get full path of export directory")
+    );
+
+    // Clear the `src/bindings` directory
+    if export_path.exists() {
+        println!("Clearing old bindings...");
+        fs::remove_dir_all(export_path).expect("Failed to clear bindings directory");
+    }
+
+    fs::create_dir_all(export_path).expect("Failed to recreate bindings directory");
+
+    // Generate TypeScript bindings
+    println!("Exporting TypeScript bindings...");
+
+    let result = Notebook::export_all_to(export_path);
+    if let Err(e) = result {
+        eprintln!("Failed to export TypeScript bindings: {}", e);
+        exit(1);
+    }
+
+    // Generate `index.ts` file
+    println!("Generating index.ts...");
+    let mut index_file = String::new();
+    for entry in fs::read_dir(export_path).expect("Failed to read bindings directory") {
```

Comment:
🦸

Reply:
only slightly annoying thing here is that we're going to have to list out every parent type we want generated by hand (just the notebook for now). we can probably find a better way to do this eventually, but we'll see if/when it becomes a pain